src/server.c

```diff
@@ -3538,7 +3538,10 @@ int processCommand(client *c) {
         c->cmd->proc != multiCommand &&
         c->cmd->proc != watchCommand &&
         c->cmd->proc != quitCommand &&
-        c->cmd->proc != resetCommand)
+        c->cmd->proc != resetCommand &&
+        c->cmd->proc != bgrewriteaofCommand &&
+        c->cmd->proc != bgsaveCommand &&
+        c->cmd->proc != saveCommand)
```

Comment:
I don't think this fix is right.
The commands that we allow not to be queued, are ones that alter the client state (abort or execute the transaction).

Maybe the right approach is to error on these and just disallow them in a transaction (not sure if it'll break any app or test).

Reply:
Yes, I also tend to disallow them, but I'm not sure if anyone has already relied on this behavior. After all, this will not cause data errors in previous versions. and we also have tests that rely on this behavior.
```
test {BGREWRITEAOF is refused if already in progress} {
        catch {
            r multi
            r bgrewriteaof
            r bgrewriteaof
            r exec
        } e
        assert_match {*ERR*already*} $e
        while {[string match {*aof_rewrite_scheduled:1*} [r info persistence]]} {
            after 100
        }
    }
```

Reply:
i think we can ignore that test, it isn't really making use of the transaction, it could have used a pipeline just the same (write the two commands to the socket at once).
or add a `key-save-delay` config.

I think these commands don't make sense inside a transaction, and now is the right time to change that (introduce a change in a major version).
the only other alternatives are:
1. execute them before the transaction, like this PR does (which i think violates the transaction (queued commands that where received before them will execute after)
2. queue these to be executed after the transaction (also seem to violate the user intent)
3. make the fork in the middle of the transaction (violates it's atomicity)